File: problems/0014-longest-common-prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] arr) {
        int max = arr[0].length();
        for (int x = 1; x < arr.length; x++) {
            if (arr[x].length() < max) {
                max = arr[x].length();
            }
        }
        StringBuilder out = new StringBuilder();
        for (int x = 0; x < max; x++) {
            char c = arr[0].charAt(x);
            boolean a = true;
            for (String s : arr) {
                if (c != s.charAt(x)) {
                    a = false;
                    break;
                }
            }
            if (a) {
                out.append(c);
            } else {
                break;
            }
        }
        return(out.toString());
    }
}

File: problems/0014-longest-common-prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int l = 0;
        for (int i = 0; i < strs[0].size(); i++) {
            char c = strs[0][i];
            for (int j = 1; j < strs.size(); j++) {
                if (strs[j][i] != c) {
                    return strs[0].substr(0, l); 
                }
            }
            l++;
        }
        return strs[0].substr(0, l);
    }
};